internal/report/synth.go

```diff
@@ -0,0 +1,36 @@
+package report
+
+import (
+	"github.com/google/pprof/profile"
+)
+
+// synthCode assigns addresses to locations without an address.
+type synthCode struct {
+	next uint64
+	addr map[*profile.Location]uint64 // Synthesized address assigned to a location
+}
+
+func newSynthCode(mappings []*profile.Mapping) *synthCode {
+	// Find a larger address than any mapping.
+	s := &synthCode{next: 1}
+	for _, m := range mappings {
+		if s.next < m.Limit {
```

Comment:
I saw in the past profiles with mapping.Limit == ~0ULL set, but I think the combination of having both zero address locations AND such fake mapping in the same profile is very unlikely, so we are probably good.